### ISSUE TITLE
[2.5.9] "Could not delete local mailbox during move of ..." ... sometimes

### ISSUE THREAD

**Issue by MASHtm** [cyrusimap/cyrus-imapd#48]
Since I updated my backends from 2.5.8 to 2.5.9 some of them complain with
"Could not delete local mailbox during move of user..." at the end of an xfer.

This happens in imapd.c:xfer_delete()->mboxlist.c:mboxlist_deletemailbox()

Looking at some strace/ltrace outputs an comparing them with backends which do not complain I'm pretty sure it is
```
    r = mboxlist_lookup_allow_all(name, &mbentry, NULL);
    if (r) goto done;
```
which returns IMAP_MAILBOX_NONEXISTENT. I modified xfer_delete to print the error code.

Is it possible that the recent changes in twoskip opened a race condition? Since the same backend does it either always or never it seems to depend on the database and its contents.

Compared to 2.4 which uses:
```
    r = mboxlist_mylookup(name, &mbentry, NULL, 1);
 ```
2.5 uses
```
    r = mboxlist_lookup_allow_all(name, &mbentry, NULL);
```
which is a wrapper for
```
    return mboxlist_mylookup(name, entryptr, tid, 0);
```

The difference is that 2.4 used a writelock, 2.5 does not.

**Comment by MASHtm** [comment]
cyr_dbtool ... repack
does not fix it if it fails

**Comment by MASHtm** [comment]
using 
mboxlist_mylookup(name, &mbentry, NULL, 1);
instead of 
mboxlist_lookup_allow_all(name, &mbentry, NULL);
doesn't fix it.

Converting mailboxes.db to skiplist doesn't fix it either.

This bug reaches critical dimension in our infrastructure. Every help is appreciated.

**Comment by MASHtm** [comment]
this one is caused by a local patch ... sorry for the noise. I tried to port
https://bugzilla.cyrusimap.org/show_bug.cgi?id=3844
to 2.5. The IMAP_MAILBOX_NONEXISTENT is returned by mailbox_open_iwl() because I added the mboxlist_lookup() in front of the mboxname_lock().

2.4 returned MBTYPE_DELETED searching with mboxlist_lookup(). 2.5 does not. using mboxlist_lookup_allow_all() instead fixes this bug.
